syncd/scripts/syncd_init_common.sh

```diff
@@ -24,12 +24,20 @@ fi
 # Use temporary view between init and apply
 CMD_ARGS+=" -u"
 
-case "$(cat /proc/cmdline)" in
-  *fast-reboot*)
+BOOT_TYPE="$(cat /proc/cmdline | grep -o 'SONIC_BOOT_TYPE=\S*' | cut -d'=' -f2)"
```

Comment:
this can cause compatibility issue. image you are on old image and want to fastfast boot into new image. We need to be aware of this issue.